client/src/components/MultiJurisdictionAudit/AASetup/Contests/HybridContestForm.test.tsx

```diff
@@ -0,0 +1,196 @@
+import React from 'react'
+import { Route } from 'react-router-dom'
+import { fireEvent, screen, waitFor } from '@testing-library/react'
+import userEvent from '@testing-library/user-event'
+import {
+  renderWithRouter,
+  withMockFetch,
+  regexpEscape,
+} from '../../../testUtilities'
+import relativeStages from '../_mocks'
+import Contests from '.'
+import { aaApiCalls } from '../../_mocks'
+import { IContest, INewContest } from '../../useContestsBallotComparison'
+import { hybridContestsInputMocks } from './_mocks'
+
+const { nextStage, prevStage } = relativeStages('target-contests')
+
+const render = (isTargeted: boolean = true) =>
+  renderWithRouter(
+    <Route path="/election/:electionId/setup">
+      <Contests
+        auditType="HYBRID"
+        locked={false}
+        isTargeted={isTargeted}
+        nextStage={nextStage}
+        prevStage={prevStage}
+      />
+    </Route>,
+    { route: '/election/1/setup' }
+  )
+
+const apiCalls = {
+  getStandardizedContests: {
+    url: '/api/election/1/standardized-contests',
+    response: [
+      {
+        name: 'Contest 1.\'"', // Make sure dots and quotes in the name work
+        jurisdictionIds: ['jurisdiction-id-1', 'jurisdiction-id-2'],
+      },
+      {
+        name: 'Contest 2',
+        jurisdictionIds: ['jurisdiction-id-1'],
+      },
+      { name: 'Contest 3', jurisdictionIds: ['jurisdiction-id-2'] },
+    ],
+  },
+  getContests: (contests: IContest[]) => ({
+    url: '/api/election/1/contest',
+    response: { contests },
+  }),
+  putContests: (contests: INewContest[]) => ({
+    url: '/api/election/1/contest',
+    options: {
+      method: 'PUT',
+      body: JSON.stringify(contests),
+      headers: { 'Content-Type': 'application/json' },
+    },
+    response: { status: 'ok' },
+  }),
+  submitContests: (mockContest: object) => ({
+    url: '/api/election/1/contest',
+    options: {
+      method: 'PUT',
+      body: JSON.stringify(mockContest),
+      headers: { 'Content-Type': 'application/json' },
+    },
+    response: { status: 'ok' },
+  }),
+}
+
+jest.mock('uuidv4', () => {
+  let id = 0
+  return () => {
+    id += 1
+    return id.toString()
+  }
+})
+
+function typeInto(input: Element, value: string): void {
+  // TODO: do more, like `focusIn`, `focusOut`, `input`, etc?
+  fireEvent.focus(input)
+  fireEvent.change(input, { target: { value } })
+  fireEvent.blur(input)
+}
+
+describe('Audit Setup > Contests (Hybrid)', () => {
+  const newContest = [
+    {
+      id: '1',
+      name: 'Contest 1.\'"',
+      isTargeted: true,
+      totalBallotsCast: 30,
+      numWinners: 1,
+      votesAllowed: 1,
+      jurisdictionIds: [],
```

Comment:
The frontend should submit the jurisdiction ids for the new contest based on the standardized contests, similar to how we do it in ballot comparison.

Reply:
@jonahkagan if I'm tracking right, then, we would need to include something like the following (from the `ContestSelect` component for Ballot Comparison audits) for Hybrid audits:

```
const filteredStandardizedContests = standardizedContests.filter(
    ({ name, jurisdictionIds }) => {
      const jurisdictionNames = jurisdictionIds.map(
        jurisdictionId =>
          jurisdictions.find(({ id }) => id === jurisdictionId)!.name
      )
      return [name, ...jurisdictionNames].some(str =>
        str.toLowerCase().includes(filter.toLowerCase())
      )
    }
  )
```

Am I tracking right?

Reply:
I think that code snippet is for the search bar in ContestSelect.

We need something that looks up the jurisdiction ids from the standardized contests and attaches them to the new contest object before submitting it to the server. Another way to think about it would that even though we hide the contest universe field from the user, we still need to submit the contest universe the the server, and the contest universe should be set when the user selects a contest from the dropdown.

Reply:
Lol, you're right. I read that wrong.

Gotcha though, that makes sense. Thanks!